### ISSUE TITLE
FR: Add MeshChat 2.x package for Raspberry Pi

### ISSUE THREAD

**Issue by csadams** [kn6plv/meshchat#22]
Feature Request...
Make a version of MeshChat 2.x for Raspberry Pi.  

Many/most of the Mesh Chat servers run on a Raspberry Pi, so that they can more easily provide extra storage, and move the service computing off of the mesh node.  They currently have to run the old version 1.02 on the Pi (and API v2.x on the node) to make it work.  Would be great to get everyone on 2.x.

**Comment by csadams** [comment]
And yes, I know this is a big request... just getting it recorded for posterity per request from KN6PLV.  :-)

**Comment by hickey** [comment]
Actually this can be done today by installing the API package on a node and then setting the config file up to pull data from the node with the API package. I don't think it is well documented, but certainly it works. 

I have another project that will be announced in another day or two that is somewhat similar where I need information from a node. I have been thinking about building a standard API daemon that will provide the information out of OLSR for non-node services. So this may be a bit easier in the future.

**Comment by hickey** [comment]
Actually I was just thinking about this issue again and realized that there is a portion of this issue that is actionable. There is one thing that does not happen today but probably will be happening in the future is automatic package builds. 

I currently have everything running in a branch for the OpenWRT package format so that MeshChat can be installed on a node. I was hoping to tidy up the branch this past weekend, but a couple of family emergencies took priority. So hopefully that branch will be ready by the end of the week for a code review. 

One of the planned enhancements once that branch is merged is to start producing Debian packaging. Then one could download a MeshChat package and install it directly on a Raspberry Pi or other Linux distribution that uses the Debian package manager.

**Comment by csadams** [comment]
Yes, we know that MeshChat works on a Raspberry Pi by using ver 1.02 on the RPi, and 2.x API package on the node... I'm running my server like that, as are numerous others on the network here in the Bay Area.  But this is about making a v2.x package for Raspberry Pi / debian.  

Like you said, 2-version combo currently required for RPis is not well documented (or not documented at all), and it's very confusing for newcomers to have to run one version of the server, and another version on the node.  Plus, v1.02 for debian does not include the bug fixes and other enhancements available in v2.x.  Since we are recommending to host MeshChat on a Pi whenever possible (and not on a node) primarily for the ease of adding file storage, it would be really nice if it was all just ver 2.x all around, and we could leave the old v1.02 and it's outdated documentation behind. (I'm thinking of writing some new docs for v2.x).  In discussing with Tim, he agreed that there should eventually be a v2.x for debian, and asked me to file this request, so hopefully we can keep it open.  If I'm understanding  your second comment right, sounds like your planned package building setup may solve this a lot sooner than I had dreamed... Thanks!

**Comment by csadams** [comment]
Looks like this (v2.x package for debian / RPi) was also discussed here: 
https://github.com/kn6plv/meshchat/pull/15#issuecomment-1596291031

**Comment by hickey** [comment]
I believe that the LUA version of MeshChat (i.e. v2.x) works on Raspberry Pi as is. One does need to install additional packages (namely LUA) to get it working correctly.

**Comment by csadams** [comment]
Oh interesting.  Tim hadn't mentioned that.  I thought it needed a .deb package, but if there's a way to insatall the .ipk on an RPi, that's great.  My linux knowledge is limited, so I'll need to do some research.  Let me know if there are any additional pointers for how to do that, and I might give it a try.

**Comment by hickey** [comment]
You can unpack the .ipk file to get the individual files out (or just as easy to copy the files out of the repository). There may be some utility that you can run on Debian to install the .ipk files directly onto the file system--I have not looked for anything. It might be just as easy to build the 'opkg' binary on a Raspberry Pi. The real solution is to start building native Debian packages, but that will not really happen until the repo reorganization is complete.

**Comment by hickey** [comment]
BTW, This is what I had done on a VM that is running Debian a while back. I just pulled the files out of the repo, put them in the proper place on the VM and got LUA installed. I don't think that there were any other packages I had to install, but I can not be 100% certain at the moment. Everything just comes up and works. The one caveat is that the `platform` variable needs to be updated and there is another variable to set for the MeshChat API location, but I don't have access to the node right now to see what the variable name is.
